controllers/ovs/flowsconfig_types.go

```diff
@@ -0,0 +1,66 @@
+package ovs
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/netobserv/network-observability-operator/api/v1alpha1"
+)
+
+type flowsConfig struct {
+	v1alpha1.FlowCollectorIPFIX `json:",inline"`
+	SharedTarget                string `json:"sharedTarget,omitempty"`
+	NodePort                    int32  `json:"nodePort,omitempty"`
+}
+
+// TODO: for better maintainability, implement map-to-struct/struct-to-map using reflection
```

Comment:
I'm surprised there's nothing like that in kube client/api to deal with config maps.
Alternatively we could use this: https://pkg.go.dev/github.com/mitchellh/mapstructure?utm_source=godoc

Reply:
Yeah, the pain here is that it's not just about serializing/deserializing as map, but also that the configmap data is in the form `map[string]string` and it requires to convert from string to `int` in some fields.

Reply:
I haven't tried it but I think `mapstructure` lib can deal with non-string fields too, nested fields, etc.

Reply:
I tried it and worked like a charm when passing from `map[string]string` --> `flowsConfig`. It required extra tweaking when passing from `flowsConfig` --> `map[string]string` but the result is anyway much better than hardcoding the struct fields.